cmd/plugin/builder/plugin/build-packages.go

```diff
@@ -57,12 +57,25 @@ func (bpo *BuildPluginPackageOptions) BuildPluginPackages() error {
 					continue
 				}
 
+				// Check whether the binary exec file is valid
+				valid, err := helpers.ValidatePluginBinary(pluginBinaryFilePath)
+
+				// Continue to build other plugins and skip if validation on current plugin binary fails
+				if err != nil {
+					continue
+				}
+
+				// Continue to build other plugins and skip if validation on current plugin binary fails
+				if !valid {
+					continue
+				}
+
```

Comment:
We should return an error here if the binary file exists but it is not a valid file.

In the previous step, we skip it and continue because it is possible that the user has not provided the plugin binary for one of the os_arch combinations. But if binary exists we should throw an error if validation fails.

Reply:
makes sense. Updated to return err if validation fails